routes/admin.js

```diff
@@ -0,0 +1,280 @@
+const keys = require("../config/nodeKeys");
+const { spawn } = require('child_process');
+const passport = require("passport");
+const flash = require("express-flash");
+const session = require("express-session");
+const path = require('path');
+const bcrypt = require('bcrypt');
+
+module.exports = (app, pool) => {
+
+  /* Configure Passport, the login mechanism for the admin page */
+  const initializePassport = require("../initializePassport");
+  initializePassport(passport, pool);
+  app.use(
+    session({
+      secret: 'secret',
+      resave: false,
+      saveUninitialized: false
+    })
+  );
+  app.use(passport.initialize());
+  app.use(passport.session());
+  app.use(flash());
+  app.use(function (req, res, next) {
```

Comment:
So if I understand correctly this is the middleware to handle errors?

Reply:
...just trying to understand.  We should probably make sure that that we have a `res.status(500)` somewhere in the error handling.

Reply:
Man, github doesn't make it easy to see what code is being commented on! Took me a hot second to find and see the complete code :) 

So, the answer is both yes and no. IIUC, express-flash is specifically for showing messages to the user, based on their input, and doesn't map to (or handle) server-side errors at all. So we do definitely need some true error handling! 

For context, the express-flash messages function a lot like input error messages you'd get in an html form in the browser; Examples of flash ""error"" messages would be "that email isn't registered," "passwords do not match" etc. They're not true errors from the code side of things, just mistakes the user needs to fix. 

Overall, I definitely hear you about (actual code) error-handling! That's definitely something we need to fix. I'll put that on the agenda for the meeting today, to figure out how we want to do that. 

tl;dr - express-flash messages are for display only, to the user. They can be buggy in and of themselves, and aren't totally necessary, so we can think about either getting rid of them, or making sure we iron out the kinks before going live. And we definitely need better error handling in general, so let's talk about the best way to do that :)

Reply:
thank your for the context.